packages/integration-sdk-core/src/data/createIntegrationEntity.ts

```diff
@@ -179,6 +179,52 @@ function generateEntity({
   return entity;
 }
 
+/**
+ * Validates that the provided value conforms to the supported types.
+ * If the value is an array, this function is called recursively on each element
+ * to ensure nested arrays are also validated. This function throws an error
+ * if it encounters a value type that is not supported.
+ *
+ * @param value The value to be validated. It can be a single value or an array.
+ *              For arrays, each element is validated recursively.
+ * @param path The path to the current property being validated, used for error messaging.
+ *             This is updated with each recursive call to reflect the current context.
+ */
+export function validateValueType(
+  value: any,
+  path: string,
+  depth: number = 1,
```

Comment:
nit: `depth` is missing from the jsdoc. Consider a rename to `currentValueDepth`